### ISSUE TITLE
Error with using allowJs and importing Javascript file

### ISSUE THREAD

**Issue by raybooysen** [TypeStrong/ts-loader#171]
Using the following example repository.  
https://github.com/raybooysen/ts-loader-example

If you `npm install` and then run `webpack`, webpack will output the following error:

```
ERROR in C:\source\ts-loader-example\tsconfig.json
error TS5055: Cannot write file 'C:/source/ts-loader-example/src/ts-loader-example/myId.js' because it would overwrite input file.

ERROR in ./src/ts-loader-example/headerBar.tsx
Module build failed: Error: Typescript emitted no output for C:\source\ts-loader-example\src\ts-loader-example\headerBar.tsx
    at Object.loader (C:\source\ts-loader-example\node_modules\ts-loader\index.js:437:15)
```

I don't understand why this is happening.  If, in Webpack, I write TS files that only import TS files, then there are no errors and typescript is not emitting any files anywhere.  

Is this a misunderstanding on my part?

***UPDATE*** 
If I add, to my tsconfig file `"outDir": "./RANDOM"`, then everything works, but there is never a folder called RANDOM created.

My TSConfig is as follows:

```
{
    "compilerOptions": {
        "noImplicitAny": false,
        "removeComments": false,
        "preserveConstEnums": true,
        "sourceMap": true,
        "noEmitOnError": true,
        "noEmit": false,
        "allowJs": true,
        "jsx": "preserve",
        "target": "es2015",
        "module": "es2015"
    },
    "files": [
        "typings/typings.d.ts"
    ]

}
```

webpack.config.js
```
var path = require('path');
var webpack = require('webpack');

var babelQuery = JSON.stringify({
  presets: ['es2015', 'react']
});


module.exports = {
  entry: [
    './src/ts-loader-example/index.js'
  ],
  output: {
    path: './build/',
    filename: 'ts-loader-example.js',
    libraryTarget: "umd",
    library:"ts-loader-example",
  },
  
  debug: true,
  //devtool: 'source-map',
  module: {

    loaders: [
      {
        test: /\.jsx?$/,
        include: path.join(__dirname, 'src'),
        loader: 'babel-loader',
        query: {
          presets: ['es2015', 'react']
        }
      },
      {
        test: /\.ts(x?)$/,
        loader: 'babel?' + babelQuery + '!ts-loader'
      }
    ]
  }
};
```

**Comment by tqmukas** [comment]
There is a bug in ts-loader. Your entry file has to be always a *.ts file, otherwise it won't work even when `allowJs` is set to `true`.

I registered an issue already: [#163](https://github.com/TypeStrong/ts-loader/issues/163)

**Comment by jbrantly** [comment]
Thanks @tqmukas

Closing this as a duplicate.

**Comment by blakeembrey** [comment]
@jbrantly Actually, I'm pretty sure this is a different issue since I walked him through it. When there's no out file or out directory, the TypeScript compiler is erroring because the `.js` file input would override the `.js` output. Obviously this isn't the case because everything is in memory, so I'd recommend setting phantom `out` and `outDir` by default to avoid this issue.

**Comment by jbrantly** [comment]
Whoops. I didn't actually triage in depth myself. Thanks!

**Comment by sirwolfgang** [comment]
What is the timeline for the next release?

**Comment by s-panferov** [comment]
@blakeembrey @jbrantly there is another way to solve this, see https://github.com/Microsoft/TypeScript/issues/7363

**Comment by jbrantly** [comment]
@s-panferov Thanks a bunch for the tip!

**Comment by thomasboyt** [comment]
So, is the "right" way to solve this issue to have ts-loader use the `suppressOutputPathCheck` option? It's an internal option so it can't be set in tsconfig.json and would have to be set by ts-loader. I don't know if there are any potential negative side effects to this, but certainly it seems a lot nicer than setting a fake "outDir" :)

**Comment by thomasboyt** [comment]
Actually, unless I'm missing something, with `suppressOutputPathCheck` being an internal option there's no way for ts-loader to even pass it to the compiler. Maybe I'm missing something, though.

**Comment by s-panferov** [comment]
@thomasboyt, hm, this option works quite well in [`awesome-typescript-loader`](https://github.com/s-panferov/awesome-typescript-loader/blob/f5105f6bb80921a68234cb63556b5aeff4b2a43c/src/instance.ts#L191), I even have some working tests with Salsa support and they fail when I disable this option. So it seems that it works. What compiler version are you using for test?

**Comment by vyorkin** [comment]
@s-panferov @thomasboyt I'm using `ts-loader@0.8.2` + `typescript@next`:
```
error TS5023: Unknown compiler option 'suppressOutputPathCheck'.
```

**Comment by johnnyreilly** [comment]
This should be fixed with release 0.9.3
